dspace-api/src/main/java/org/dspace/content/dao/impl/RelationshipDAOImpl.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.content.dao.impl;
+
+import java.sql.SQLException;
+import java.util.List;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+import org.dspace.content.Item;
+import org.dspace.content.Relationship;
+import org.dspace.content.Relationship_;
+import org.dspace.content.dao.RelationshipDAO;
+import org.dspace.core.AbstractHibernateDAO;
+import org.dspace.core.Context;
+
+public class RelationshipDAOImpl extends AbstractHibernateDAO<Relationship> implements RelationshipDAO {
+
+    public List<Relationship> findByItem(Context context, Item item) throws SQLException {
+        CriteriaBuilder criteriaBuilder = getCriteriaBuilder(context);
+        CriteriaQuery criteriaQuery = getCriteriaQuery(criteriaBuilder, Relationship.class);
+        Root<Relationship> relationshipRoot = criteriaQuery.from(Relationship.class);
+        criteriaQuery.select(relationshipRoot);
+        criteriaQuery
+            .where(criteriaBuilder.or(criteriaBuilder.equal(relationshipRoot.get(Relationship_.leftItem), item),
+                                      criteriaBuilder.equal(relationshipRoot.get(Relationship_.rightItem), item)));
+        return list(context, criteriaQuery, true, Relationship.class, -1, -1);
+    }
+
+    public int findLeftPlaceByLeftItem(Context context, Item item) throws SQLException {
+        CriteriaBuilder criteriaBuilder = getCriteriaBuilder(context);
+        CriteriaQuery criteriaQuery = getCriteriaQuery(criteriaBuilder, Relationship.class);
+        Root<Relationship> relationshipRoot = criteriaQuery.from(Relationship.class);
+        criteriaQuery.select(relationshipRoot);
+        criteriaQuery.where(criteriaBuilder.equal(relationshipRoot.get(Relationship_.leftItem), item));
+        List<Relationship> list = list(context, criteriaQuery, true, Relationship.class, -1, -1);
+        list.sort((o1, o2) -> o2.getLeftPlace() - o1.getLeftPlace());
+        if (!list.isEmpty()) {
+            return list.get(0).getLeftPlace();
+        } else {
+            return 1;
```

Comment:
Shouldn't this `return 0` or `list.get(0).getLeftPlace() + 1` (and remove the + 1 in `MetadataImport.buildRelationObject()`)? Or is there any reason to skip the place 1?

Reply:
The current implementation of the place columns was kept simple
We’re currently redesigning this functionality to support mixing entities and plain text values. At the time of the initial implementation, this was still unclear. The new implementation should create a behavior much more similar to the place column in metadata as well